src/LEGO.AsyncAPI.Bindings/Sqs/Principal.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) The LEGO Group. All rights reserved.
+
+namespace LEGO.AsyncAPI.Bindings.Sqs;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using LEGO.AsyncAPI.Models.Interfaces;
+using LEGO.AsyncAPI.Readers.ParseNodes;
+
+public class Principal : IAsyncApiElement
+{
+    public Principal(IPrincipalValue value)
+    {
+        this.Value = value;
+    }
+
+    public IPrincipalValue Value { get; set; }
```

Comment:
consider making the setter private.

Reply:
I think it may need to stay public in order to call serialise from the statement class https://github.com/LEGO/AsyncAPI.NET/pull/187/files#diff-cf6ebfabac787b2843227d8ef7643d63740a56eb7e2bf6b9e0d88f9054dcad5bR49

Reply:
no no, only the setter. 
```suggestion
    public IPrincipalValue Value { get; private set; }
```